### ISSUE TITLE
Add scroll_following to ScrollContainer

### ISSUE THREAD

**Issue by rcorre** [godotengine/godot-proposals#3629]
### Describe the project you are working on

A multiplayer game with in-game chat. When chat messages come in, the chat box should scroll to the bottom to display the most recent message.

### Describe the problem or limitation you are having in your project

Having a `ScrollContainer` scroll to the newest/bottom-most element is surprisingly difficult. First, there's not an obvious way to get the "bottom" scroll value. You either have to say `	get_v_scrollbar().value = get_v_scrollbar().max_value`

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a `scroll_following` property to `ScrollContainer` that works like [RichTextLabel.scroll_following](https://docs.godotengine.org/en/3.3/classes/class_richtextlabel.html#class-richtextlabel-property-scroll-following).

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Worst case scenario, `autoscroll` just hides the `yeild(get_tree(), "idle_frame"); scroll_vertical = 100000` logic behind the scenes. I'm hoping that maybe we can do it in a cleaner way by mimicking RichTextLabel (https://github.com/godotengine/godot/blob/2d118bd8b881fe9658e70eb8dc4fa7a6efac41a3/scene/gui/rich_text_label.cpp#L2187-L2189)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be worked around with two lines of script:

```
yield(get_tree(), "idle_frame")
scroll_vertical = 9223372036854775807 # is there an INT_MAX constant?
```

However, those lines are very non-obvious. If we think this is the right approach though, I can just document it instead.

### Is there a reason why this should be core and not an add-on in the asset library?

The solution is simple, but non-obvious.

**Comment by pycbouh** [comment]
I think this is already achievable with `ensure_control_visible()`. The problem is that this method only works on direct children. I think it can be modified to work with indirect children as well. It may still require waiting for `idle_frame` though.

**Comment by KoBeWi** [comment]
Isn't `follow_focus` enough? You could do
```
add_child(something)
something.grab_focus()
```

**Comment by pycbouh** [comment]
Maybe you don't want to change focus. Either way, though, I think it is solvable as is.

**Comment by KoBeWi** [comment]
Why wouldn't you want to change focus if the new item is important and you want to scroll to it?

**Comment by pycbouh** [comment]
> Why wouldn't you want to change focus if the new item is important and you want to scroll to it?

Well, if you have a chat app, you want the last messages, coming from the bottom, to always be visible, but you don't want to steal the focus from the text input field. Something like that I assume is the use case for OP.

**Comment by rcorre** [comment]
In my case, it is a non-interactable chat window shown to the side of the screen during gameplay, so it doesn't have "focus" to begin with. I guess I could try "focusing" a label anyways, maybe it wouldn't hurt anything.

Still, I think @pycbouh has a good example of where that wouldn't work.

https://github.com/godotengine/godot-proposals/issues/3629#issuecomment-985525186 is better than what I had, but the `yield` is still pretty non-obvious. I suppose just doc'ing that is better than nothing though.

**Comment by rcorre** [comment]
FWIW, I tried setting `follow_focus=True`, and calling `label.grab_focus()` after adding, and that seems to do nothing, even if I yield to `idle_frame`.

**Comment by KoBeWi** [comment]
Does your label have focus mode different than NONE? You should get a warning about that.

**Comment by rcorre** [comment]
You're right, there was a warning, and it does work with `FOCUS_ALL`. You still need the `yield` trick though.
